rust/src/transformations/make_stable_expr/expr_boolean_function/mod.rs

```diff
@@ -81,13 +81,12 @@ where
         data_column.nullable = false;
     }
 
-    if matches!(bool_function, Not) && data_column.dtype() != DataType::Boolean {
-        // under these conditions, the expression performs a bitwise negation
-        data_column.drop_bounds()?;
+    data_column.set_dtype(if matches!(bool_function, Not) {
+        // under these conditions, the expression performs a bitwise negation and all descriptors are dropped
+        data_column.dtype()
```

Comment:
So just by itself this statement:
```
data_column.set_dtype(data_column.dtype())
```
would drop descriptors? I'm a little surprised that it changes anything. Might be working correctly, but the explicit `drop_bounds` in the previous state seems more clear.